src/taskgraph/run-task/run-task

```diff
@@ -666,7 +666,7 @@ def git_checkout(
 
     if ref:
         args.extend(["-B", ref])
-    args.append(commit if commit else ref)
+    args.append(commit if commit else "FETCH_HEAD")  # Set by git fetch
```

Comment:
I'd suggest we provide a little bit more context for future readers. How about something like:

```suggestion
    # The last `git fetch` didn't set any remote repo it just reset this `FETCH_HEAD` commitish
    args.append(commit if commit else "FETCH_HEAD")
```

?